### PR TITLE
Add the support of JSON.TYPE command

### LINKED ISSUE #1826: Add support for the JSON.TYPE command
Please refer to #1805.

### PR DESCRIPTION
**Pull request by git-hulk**
This closes https://github.com/apache/kvrocks/issues/1826

For the JSON type mapping, please refer: https://github.com/RedisJSON/RedisJSON/blob/ecede3c314851ccbc793d976c890bf9ea291a371/redis_json/src/key_value.rs#L378

### CODE REVIEW

File: src/commands/cmd_json.cc [hunk 1]
```diff
@@ -123,8 +123,32 @@ class CommandJsonArrAppend : public Commander {
...
+ public:
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::Json json(svr->storage, conn->GetNamespace());
+
+    std::vector<std::string> types;
+
+    std::string path = "$";
+    if (args_.size() > 2) {
+      path = args_[2];
+    }
```
Comment (PragmaTwice) [inline src/commands/cmd_json.cc]:
```suggestion
    if (args_.size() == 2) {
      path = args_[2];
    } else if (args_.size() > 2) {
      return {Status::RedisExecErr, "The number of arguments is more than expected"};
    }
```

File: src/types/json.h [hunk 2]
```diff
@@ -132,6 +132,46 @@ struct JsonValue {
...
+            break;
+          case jsoncons::json_type::bool_value:
+            types->emplace_back("boolean");
+            break;
+          case jsoncons::json_type::int64_value:
+            types->emplace_back("integer");
+            break;
+          case jsoncons::json_type::uint64_value:
+            types->emplace_back("integer");
+            break;
```
Comment (PragmaTwice) [inline src/types/json.h]:
```suggestion
          case jsoncons::json_type::int64_value:
          case jsoncons::json_type::uint64_value:
            types->emplace_back("integer");
            break;
```

**Review by PragmaTwice — COMMENTED** [review]
Generally LGTM, just some style issue above.

File: src/commands/cmd_json.cc [hunk 3]
```diff
@@ -123,8 +123,34 @@ class CommandJsonArrAppend : public Commander {
...
+    redis::Json json(svr->storage, conn->GetNamespace());
+
+    std::vector<std::string> types;
+
+    std::string path = "$";
+    if (args_.size() == 2) {
+      path = args_[2];
+    } else if (args_.size() > 2) {
+      return {Status::RedisExecErr, "The number of arguments is more than expected"};
+    }
```
Comment (PragmaTwice) [inline src/commands/cmd_json.cc]:
```suggestion
    if (args_.size() == 3) {
      path = args_[2];
    } else if (args_.size() > 3) {
      return {Status::RedisExecErr, "The number of arguments is more than expected"};
    }
```

sorry :rofl:

Comment (git-hulk) [reply src/commands/cmd_json.cc]: aha, I also missed this.

**Review by PragmaTwice — APPROVED** [review]

File: src/types/json.h [hunk 4]
```diff
@@ -132,6 +132,44 @@ struct JsonValue {
...
+          case jsoncons::json_type::bool_value:
+            types->emplace_back("boolean");
+            break;
+          case jsoncons::json_type::int64_value:
+          case jsoncons::json_type::uint64_value:
+            types->emplace_back("integer");
+            break;
+          case jsoncons::json_type::double_value:
+            types->emplace_back("number");
```
Comment (mapleFU) [inline src/types/json.h:151]: Should we support `half` here?

File: src/types/json.h [hunk 5]
```diff
@@ -132,6 +132,44 @@ struct JsonValue {
...
+            break;
+          case jsoncons::json_type::string_value:
```
Comment (mapleFU) [inline src/types/json.h:153]: So binary is not supported?

Comment (PragmaTwice) [reply src/types/json.h:153]: I think it is not a type in JSON documents.

File: src/types/json.h (hunk 4 above)
Comment (PragmaTwice) [reply src/types/json.h:151]: ditto https://github.com/apache/kvrocks/pull/1847#discussion_r1367741127

**Pull request merged**